### PR TITLE
run_e3sm script: Replace default case name and case group placeholders

### PR DESCRIPTION
**Pull request by sarats**
Replace default case name and case group placeholders.
This is required to avoid unwarranted usage of case_group that
interferes with aggregation of various production simulation campaigns.

[BFB]

Testing:
I was able to run ./create_newcase with --case-group "" as defined.

### CODE REVIEW

**Review by rljacob — APPROVED** [review]

**Comment by sarats** [comment]
Looks like run_e3sm script would not pass the empty string to create_newcase invocation. So, I had to conditionally include the -case-group argument when it is defined.

**Comment by rljacob** [comment]
We should do this on maint-2.0 as well.

File: run_e3sm.template.sh [hunk 1]
```diff
@@ -24,8 +24,10 @@ readonly PROJECT="e3sm"
 # Simulation
 readonly COMPSET="WCYCL1850"
 readonly RESOLUTION="ne30pg2_EC30to60E2r2"
-readonly CASE_NAME="v2.LR.piControl"
-readonly CASE_GROUP="v2.LR"
+# TODO: CHANGE the following CASE_NAME to desired values
```
Comment (forsyth2) [inline run_e3sm.template.sh:27]: Is this an actual TODO for the developers/maintainers of `run_e3sm.template.sh` or is it a direction for people who copy this template? If the latter, something like ` # IMPORTANT` or `# BEFORE RUNNING` might be better.

File: run_e3sm.template.sh [hunk 2]
```diff
@@ -245,18 +247,33 @@ create_newcase() {
...
+			--compset ${COMPSET} \
+			--res ${RESOLUTION} \
+			--machine ${MACHINE} \
+			--project ${PROJECT} \
+			--walltime ${WALLTIME} \
+			--pecount ${PELAYOUT}
+	else
+		${CODE_ROOT}/cime/scripts/create_newcase \
+			--case ${CASE_NAME} \
+			--case-group $(CASE_GROUP) \
```
Comment (forsyth2) [inline run_e3sm.template.sh:265]: Why the change from `{}` to `()`?

**Review by forsyth2 — APPROVED** [review]
I didn't run myself, but the code changes seem reasonable. I just made a couple minor comments.

File: run_e3sm.template.sh (hunk 1 above)
Comment (sarats) [reply run_e3sm.template.sh:27]: It's for the user to do. I can change to "BEFORE RUNNING"

File: run_e3sm.template.sh (hunk 2 above)
Comment (sarats) [reply run_e3sm.template.sh:265]: That was another way for handling empty string. We don't need it now as we are doing an explicit check. I will remove.

**Comment by sarats** [comment]
Ok, I updated the instructions for user to set case before running.
